### DISCUSSION TITLE
Change RetentionPolicy from CLASS to RUNTIME

### DISCUSSION THREAD

**Discussion by WeierE1** [mapstruct/mapstruct#3528] [Q&A]
Is it possible to change the RetentionPolicy.CLASS to RetentionPolicy.RUNTIME.
I would like to get at runtime the Mapping Definition defined in the interface with @Mapper and @Mapping. 
I have multiple database classes which i have to map on corresponding DTOs and i need to convert the sorting definition that i defined in the database queries to a sorting defintion of the dto that i can use in the frontend. Therefore i need the defined mapping at runtime.

**Comment by filiphr** [comment]
We currently have #702 for this. However, I am not that keen on changing this from `CLASS` to `RUNTIME`.

Can you please give some examples of how this would help? How does the mapping definition relate to the sorting in the database queries?

**Comment by alexisprunier** [comment]
Hi,

Exactly having the same situation here.

Long story short, we are having DTO objects that are mapped into entities using mapstruct. We would like to retrieve the @Mapping set definition using reflection. Thanks to this, we could also map the properties of a spring "Pageable.Sort". The scope of this mapping could be controlled by using mapped source attributes, and only it.

This way, we could also have a unique place where we define the mapping.

Regards,
Alexis

**Reply by filiphr** [reply]
Hey @alexisprunier ,

Can you please give us a concrete example with what you want to do. Ideally with some (pseudo) code. I am familiar with the Spring Data `Pageable.Sort`, but I can't see what you can do with reflection at the moment
